libs/api/domains/health-insurance/src/lib/soap/healthInsuranceAPI.ts

```diff
@@ -172,17 +168,14 @@ export class HealthInsuranceAPI {
 
     // Add attachments from S3 bucket
     // Attachment's name need to be exactly same as the file name, including file type (ex: skra.txt)
-
-    if (
-      inputObj.attachmentsFileNames &&
-      inputObj.attachmentsFileNames.length > 0
-    ) {
+    const arrAttachments = inputObj.attachmentsFileNames
+    if (arrAttachments && arrAttachments.length > 0) {
       logger.info(`Start getting attachments`)
       const fylgiskjol: Fylgiskjol = {
         fylgiskjal: [],
       }
-      for (let i = 0; i < inputObj.attachmentsFileNames.length; i++) {
-        const filename = inputObj.attachmentsFileNames[i]
+      for (let i = 0; i < arrAttachments.length; i++) {
+        const filename = arrAttachments[i]
         const fylgiskjal: Fylgiskjal = {
           heiti: filename,
           innihald: await this.bucketService.getFileContentAsBase64(filename),
```

Comment:
The GraphQL API does/should not have access to the Application System buckets. Another reason to do this in the template API.